aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -437,6 +441,56 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
     LOG.info("Successfully renamed table from {} to {}", from, to);
   }
 
+  @Override
+  public org.apache.iceberg.Table registerTable(
```

Comment:
I would suggest while we have the conversation going, we can add a feature flag in `AwsProperties` to control this behavior

Reply:
So if this ended up not received in OSS, we can apply it as internal patch.

Reply:
Sounds good to me, will add a feature flag: `glue.force-register-table`